source/hp/dof_handler.cc

```diff
@@ -2675,47 +2689,21 @@ namespace hp
                 ExcInternalError());
       }
 
-    // finally, do the renumbering
-    // and set the number of actually
+    // finally, do the renumbering and set the number of actually
     // used dof indices
-    renumber_dofs_internal (new_dof_indices, dealii::internal::int2type<dim>());
+    policy->renumber_dofs (new_dof_indices);
 
-    // now set the elements of the
-    // number cache appropriately
-    number_cache.n_global_dofs        = next_free_dof;
-    number_cache.n_locally_owned_dofs = number_cache.n_global_dofs;
+    // now set the elements of the number cache appropriately
+    number_cache = dealii::internal::DoFHandler::NumberCache (next_free_dof);
 
-    if (dynamic_cast<const parallel::shared::Triangulation< dim, spacedim >*>
-        (&this->get_triangulation())
-        == nullptr)
-      {
-        number_cache.locally_owned_dofs
```

Comment:
i suppose those things are happening inside `internal` function now? It seems we indeed only need `next_free_dof` to set the cache right.